### PR TITLE
MGMT-8027: Replace generic events

### PR DESCRIPTION
**Pull request by arielireni**
# Assisted Pull Request

## Description
Replaced two types of events (`generic_cluster` and `generic_host`) with a concrete event type based on the context of where they are being invoked from.

This will allow querying for events by their type.

Signed-off-by: arielireni <aireni@redhat.com>

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @masayag 
/cc @nmagnezi 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -6126,8 +6126,7 @@ func (b *bareMetalInventory) V2RegisterHost(ctx context.Context, params installe
 				log.WithError(err).Errorf("failed to register host <%s> to infra-env %s due to: %s",
 					params.NewHostParams.HostID, params.InfraEnvID.String(), err.Error())
 				// TODO Need event for infra-env instead of cluster
-				eventgen.SendGenericHostEvent(ctx, b.eventsHandler, *params.NewHostParams.HostID, params.InfraEnvID,
-					err.Error(), models.EventSeverityError)
+				eventgen.SendRegisterHostToInfraEnvFailedEvent(ctx, b.eventsHandler, *params.NewHostParams.HostID, params.InfraEnvID, err.Error())
```
Comment (masayag) [inline internal/bminventory/inventory.go:6129]: @nmagnezi do we still need the comment above ?

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -6147,8 +6146,7 @@ func (b *bareMetalInventory) V2RegisterHost(ctx context.Context, params installe
 		uerr := errors.Wrap(err, fmt.Sprintf("Failed to register host %s", hostutil.GetHostnameForMsg(host)))
 
 		// TODO event for infra-env
-		eventgen.SendGenericHostEvent(ctx, b.eventsHandler, *params.NewHostParams.HostID, params.InfraEnvID,
-			uerr.Error(), models.EventSeverityError)
+		eventgen.SendHostRegistrationFailedEvent(ctx, b.eventsHandler, *params.NewHostParams.HostID, params.InfraEnvID, uerr.Error())
```
Comment (masayag) [inline internal/bminventory/inventory.go:6149]: @nmagnezi do we still need the comment above ?

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -1438,35 +1438,33 @@ func (b *bareMetalInventory) createAndUploadNewImage(ctx context.Context, log lo
 	if err := b.updateImageInfoPostUpload(ctx, infraEnv, infraEnvProxyHash, imageType, true, v2); err != nil {
 		return common.NewApiError(http.StatusInternalServerError, err)
 	}
-	msg := b.getIgnitionConfigForLogging(ctx, infraEnv, log, imageType)
-	eventgen.SendGenericClusterEvent(ctx, b.eventsHandler, *infraEnv.ID, msg, models.EventSeverityInfo)
-	log.Infof(msg)
+	details := b.getIgnitionConfigForLogging(ctx, infraEnv, log, imageType)
+	eventgen.SendGeneratedImageWithIgnitionConfigEvent(ctx, b.eventsHandler, *infraEnv.ID, details)
```
[the remark this answers is not in the document]
Comment (arielireni) [reply internal/bminventory/inventory.go:1442]: This ticket will handle the required change - https://issues.redhat.com/browse/MGMT-8095

**Pull request merged**